packages/ui/src/common/components/Search/SearchResultsModal.tsx

```diff
@@ -19,16 +19,21 @@ import { ModalWithDataCall } from '@/common/providers/modal/types'
 import { ThreadItemBreadcrumbs } from '@/forum/components/threads/ThreadItemBreadcrumbs'
 import { ForumRoutes } from '@/forum/constant'
 
+import { useDebounce } from '../../hooks/useDebounce'
+
 export type SearchResultsModalCall = ModalWithDataCall<'SearchResults', { search: string }>
 
 export const SearchResultsModal = () => {
   const { hideModal, modalData } = useModal<SearchResultsModalCall>()
-
   const [search, setSearch] = useState(modalData.search)
   const [activeTab, setActiveTab] = useState<SearchKind>('FORUM')
   const { forum, forumPostCount, isLoading } = useSearch(search, activeTab)
-  const pattern = useMemo(() => (search ? RegExp(escapeStringRegexp(search), 'ig') : null), [search])
-
+  const isValid = () => !debouncedSearch || debouncedSearch.length === 0 || debouncedSearch.length > 2
+  const debouncedSearch = useDebounce(search, 400)
+  const pattern = useMemo(
+    () => (isValid() && search ? RegExp(escapeStringRegexp(search), 'ig') : null),
+    [debouncedSearch]
+  )
```

Comment:
Turns out setting `pattern` to null isn't enough because no pattern results in the entire post showing. Which is confusing because it looks like the result was updated despite the message saying that the search is too short.

[Screencast from 10-18-2022 05:33:12 PM.webm](https://user-images.githubusercontent.com/6571453/196480328-19ecbad5-99b0-4ae1-ba05-368bea29528b.webm)

Instead the `pattern` should remain the same:
```suggestion
  const [pattern, setPattern] = useState<RegExp | null>(null)
  useEffect(() => {
    if (isValid()) {
      setPattern(RegExp(escapeStringRegexp(debouncedSearch), 'ig'))
    }
  }, [debouncedSearch])
```